docs/en/concepts-and-designs/service-hierarchy.md

```diff
@@ -162,6 +164,22 @@ If you want to customize it according to your own needs, please refer to [Servic
   - VIRTUAL_MQ.service.name: `rabbitmq.skywalking-showcase.svc.cluster.local:5672`
   - RABBITMQ.service.name: `rabbitmq::rabbitmq.skywalking-showcase`
 
+#### KAFKA On K8S_SERVICE
+- Rule name: `short-name`
+- Groovy script: `{ (u, l) -> u.shortName == l.shortName }`
+- Description: KAFKA.service.shortName == K8S_SERVICE.service.shortName
+- Matched Example:
+  - ROCKETMQ.service.name: `kafka::kafka.skywalking-showcase`
+  - K8S_SERVICE.service.name: `skywalking-showcase::kafka.skywalking-showcase`
+
+#### VIRTUAL_MQ On KAFKA
+- Rule name: `lower-short-name-with-fqdn`
+- Groovy script: `{ (u, l) -> u.shortName.substring(0, u.shortName.lastIndexOf(':')) == l.shortName.concat('.svc.cluster.local') }`
+- Description: VIRTUAL_MQ.service.shortName remove port == KAFKA.service.shortName with fqdn suffix
+- Matched Example:
+  - VIRTUAL_MQ.service.name: `kafka.skywalking-showcase.svc.cluster.local:9092`
```

Comment:
Are both producer and consumer working for this name? Is this name showing up in the general topology?

Reply:
> Are both producer and consumer working for this name? Is this name showing up in the general topology?

kafka java plugin intercept the ```bootstrap.servers``` as peer name , so I think they  both  display the same span peer name

Reply:
Could you test that part? My concern was whether a kind of DNS resolver would make the agent plugin puts real IPs in the peer, rather than this DNS name. Is the DNS name showing up as the virtual MQ and topology/span's peer, that should be good.

Reply:
> Could you test that part? My concern was whether a kind of DNS resolver would make the agent plugin puts real IPs in the peer, rather than this DNS name. Is the DNS name showing up as the virtual MQ and topology/span's peer, that should be good.

I can, test that ,but I thnik it is similiar to pg

![image](https://github.com/apache/skywalking/assets/96932958/d4ffa511-f0bb-41d7-bb73-0e82eeabb6be)

Reply:
Please recheck. Sometime, plugins run expectedly.
Meanwhile, as you are going to.change showcase demo app, it should be the same process.

Reply:
> Please recheck. Sometime, plugins run expectedly. Meanwhile, as you are going to.change showcase demo app, it should be the same process.

Ok, I will do it on this weekend

Reply:
The case is, when Kafka is set, you may use Kafka broker addresses, and the app client is using data node addresses to communicate.
Those address may not match the one from otel monitoring and k8s part.

Reply:
> The case is, when Kafka is set, you may use Kafka broker addresses, and the app client is using data node addresses to communicate. Those address may not match the one from otel monitoring and k8s part.

May be I am not very follow your meaning ,  for end users, they should use the same address or customize script,  for showcase, we could change the otel label output, Is that so?

Reply:
My point is, from setup perspective, the producer and consumer should use Kafka service DNS name, right? Then in the real communication, is the peer in span documented as the same DNS name?

Reply:
> My point is, from setup perspective, the producer and consumer should use Kafka service DNS name, right? Then in the real communication, is the peer in span documented as the same DNS name?

Kafka span peer name is the value of the `bootstrap.servers` which user config, If the value is a DNS name, it will be a DNS name, if the value is a ip name, it will be a ip name

Reply:
OK, I see. If that is the case it should work. But let's test that to make sure it works as expected. 
It is not urgent, we still have weeks before 10.0 release(we are waiting for BanyanDB 0.6). Take your time on the weekend.